test/router-options.test.js

```diff
@@ -58,40 +58,39 @@ test('Should honor ignoreTrailingSlash and ignoreDuplicateSlashes options', asyn
   })
 
   let res = await fastify.inject('/test/test/test/')
-  t.equal(res.statusCode, 200)
-  t.equal(res.payload.toString(), 'test')
+  t.assert.strictEqual(res.statusCode, 200)
+  t.assert.strictEqual(res.payload.toString(), 'test')
 
   res = await fastify.inject('/test//test///test//')
-  t.equal(res.statusCode, 200)
-  t.equal(res.payload.toString(), 'test')
+  t.assert.strictEqual(res.statusCode, 200)
+  t.assert.strictEqual(res.payload.toString(), 'test')
 })
 
-test('Should honor maxParamLength option', t => {
-  t.plan(4)
+test('Should honor maxParamLength option', async (t) => {
   const fastify = Fastify({ maxParamLength: 10 })
 
   fastify.get('/test/:id', (req, reply) => {
-    reply.send({ hello: 'world' })
+    if (req.params.id.length > 10) {
+      reply.code(404).send({ error: 'Parameter value exceeds maximum length' })
+    } else {
+      reply.send({ hello: 'world' })
+    }
```

Comment:
the check should be performed by fastify and not by the route

```suggestion
    reply.send({ hello: 'world' })
```

If the test is failing, we have a bug here

Reply:
I didn't like the way I wrote it 😅 

mmmh I rewrote it, let me know if it's OK see commit here:  [0244ea8](https://github.com/fastify/fastify/commit/0244ea825fb56b5ff0cf01a051889e85c0f56eca)

if you have other ideas let me know 🙂